freeipa/src/main/java/com/sequenceiq/freeipa/service/stack/FreeIpaScalingValidationService.java

```diff
@@ -46,13 +43,26 @@ public void validateStackForVerticalUpscale(Stack stack, VerticalScaleRequest re
     }
 
     public void validateStackForDownscale(Set<InstanceMetaData> allInstances, Stack stack, ScalingPath scalingPath) {
-        if (allInstances.size() < AvailabilityType.HA.getInstanceCount()) {
-            LOGGER.warn("FreeIPA instance count is not allowed. Size: [{}]", allInstances.size());
-            throw new BadRequestException("Downscaling currently only available for FreeIPA installation with 3 instances");
-        }
+        validateScalingIsDownscale(scalingPath);
         executeCommonValidations(allInstances, stack, scalingPath, OperationType.DOWNSCALE);
     }
 
+    private void validateScalingIsUpscale(ScalingPath scalingPath) {
+        if (scalingPath.getOriginalAvailabilityType().getInstanceCount() > scalingPath.getTargetAvailabilityType().getInstanceCount()) {
+            throw new BadRequestException(String.format("Refusing %s as target node count is smaller than current. Current node count: %d, " +
+                            "target node count: %d.", OperationType.UPSCALE.getLowerCaseName(), scalingPath.getOriginalAvailabilityType().getInstanceCount(),
+                    scalingPath.getTargetAvailabilityType().getInstanceCount()));
+        }
+    }
+
+    private void validateScalingIsDownscale(ScalingPath scalingPath) {
+        if (scalingPath.getOriginalAvailabilityType().getInstanceCount() < scalingPath.getTargetAvailabilityType().getInstanceCount()) {
```

Comment:
shouldn't we compare with the actual original instance count here?

Reply:
This is just making sure that it is not an upscale request that was sent to the downscale endpoint.